### DISCUSSION TITLE
What's the plan RE: Godot's constant evolution?

### DISCUSSION THREAD

**Discussion by Jimmio92** [godot-jolt/godot-jolt#1023] [Questions]
First, let me start by thanking everyone involved in Jolt and godot-jolt. What a fantastic tool to give to the open source community. Thank you.

Given Godot's faster release cycle, I wonder what the plans are for godot-jolt releases; personally, I would love to use this in the dev releases though I fully understand why you'd want to only provide for stable builds of Godot. I'm a bleeding edge kind of guy, even if it means I'm drowning in it...

Do you folks start the testing/porting in the beta phase? I imagine it'd be easy to remove the version check in my own build and see where it's at compatibility wise, but I figured it'd be best to just ask those who know.

Thanks!

**Comment by mihe** [comment]
> Do you folks start the testing/porting in the beta phase?

I try to put up a beta release of my own here on GitHub when the first beta of Godot is released, yes. The release on Godot Asset Library only targets stable Godot.

> I imagine it'd be easy to remove the version check in my own build and see where it's at compatibility wise

You don't need to do a custom build per se. The version check is on the Godot side of things, and uses the values set in the `godot-jolt.gdextension` file, so just changing `compatibility_maximum` there to `4.4` will allow you to load the extension, including in dev releases.

For what it's worth, it's the aim of Godot (and its C++ bindings) that extensions be forward-compatible more or less indefinitely, and they go through some effort to at least ensure binary compatibility. So most likely you'll find that a release targeting Godot 4.3 will work fine in Godot 4.4 as well, but I don't officially support that, so if anything does break you might need to wait until I put out a proper release.
